jschon/jsonschema.py

```diff
@@ -218,21 +218,15 @@ def parentschema(self) -> Optional[JSONSchema]:
                 return parent
             parent = parent.parent
 
-    @property
+    @cached_property
```

Comment:
This makes sense. I think it implies that the `metaschema_uri` getter should also be cached.

Reply:
It turns out that `metaschema_uri` and `uri` need to be set by `"$schema"` and `"$id"` and that's why they're not cached.  Ideally, they could only be set once and cached after that.  It's still safe to cached `metaschema` because if it has not been set then it will raise an exception, and it really shouldn't ever be changed.

Reply:
Right, I see that `@cached_property` doesn't work with a setter. At the time that I modified the `parentschema` property to be cached (as well as `path` and `value` on `JSON`), I didn't do the same for `metaschema` precisely because `metaschema_uri` is updatable. Previously I've vaguely considered trying to resolve this by integrating core keywords (or at least, the 'bootstrapping' ones like `$schema` and `$id`) more tightly with `JSONSchema`, and making `JSONSchema` immutable. But that would really be overly restrictive, and in fact it was my own extension [jschon-translation](https://github.com/marksparkza/jschon-translation) that roundly dismissed the idea of making `JSONSchema` immutable. In any case, in normal usage `uri` and `metaschema_uri` are and should only be set once, and it makes sense to cache `metaschema` because it's accessed repeatedly.